types/v56/types.go

```diff
@@ -211,6 +211,7 @@ type NetworkConfiguration struct {
 	RetainNetInfoAcrossDeployments bool             `xml:"RetainNetInfoAcrossDeployments,omitempty"`
 	Features                       *NetworkFeatures `xml:"Features,omitempty"`
 	GuestVlanAllowed               *bool            `xml:"GuestVlanAllowed,omitempty"`
+	SubInterface                   *bool            `xml:"SubInterface,omitempty"`
 	DistributedInterface           *bool            `xml:"DistributedInterface,omitempty"`
```

Comment:
Is it possible to have both `SubInterface` and `DistributedInterface` set to true? If not, I'd add a comment for future reference.

Reply:
Looking at how the GUI treats these values, they should be mutually exclusive.
See the picture in [TF Issue 342](https://github.com/terraform-providers/terraform-provider-vcd/issues/342).

Reply:
Yes, I'm aware of that. However, I don't know what happens if we send both true value to API.

Reply:
A quick test with the API shows that we can't set both.
The API treats the interface types like this:
1. if both SubInterface and DistributedInterface are `nil`, it's "internal"
2. if either SubInterface or DistributedInterface are set, the corresponding interface is used.
3. If both variables are set, it's an error.

Notice that choosing the "internal" interface is the result of lack of other interfaces being set.